cmd/babylond/cmd/genhelpers/collect.go

```diff
@@ -0,0 +1,266 @@
+package genhelpers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	sdkerrors "cosmossdk.io/errors"
+
+	cfg "github.com/cometbft/cometbft/config"
+
+	checkpointingtypes "github.com/babylonlabs-io/babylon/x/checkpointing/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkruntime "github.com/cosmos/cosmos-sdk/runtime"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankexported "github.com/cosmos/cosmos-sdk/x/bank/exported"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	"github.com/cosmos/cosmos-sdk/x/genutil/types"
+)
+
+const flagGenTxDir = "gentx-dir"
+
+// CollectGenTxsCmd - return the cobra command to collect genesis transactions
+// Code coppied from https://github.com/cosmos/cosmos-sdk/blob/46be66ed4205bf109d93704dd14c0822d048972e/x/genutil/client/cli/collect.go#L22
+// The reason is that the genutil parses the gen tx message as stakingtypes.MsgCreateValidator to check balances in
+// https://github.com/cosmos/cosmos-sdk/blob/46be66ed4205bf109d93704dd14c0822d048972e/x/genutil/collect.go#L134
+func CollectGenTxsCmd(genBalIterator types.GenesisBalancesIterator, defaultNodeHome string, validator types.MessageValidator, valAddrCodec sdkruntime.ValidatorAddressCodec) *cobra.Command {
```

Comment:
as I understand this is copy paste from cosmos-sdk, so maybe we need some licence attribution on top of the file ?